### ISSUE TITLE
v0.10.0 in deep mode reports Variables not allowed in provider aws block

### ISSUE THREAD

**Issue by mdaniel** [wata727/tflint#392]
Given `main.tf` of:
```hcl
variable aws_region {
  default = "us-west-2"
}
provider aws {
  version = "~> 2.21"
  region  = var.aws_region
}
data aws_availability_zones az_list {}
```
Then running `tflint  --deep` yields a kaboom that `terraform` itself says is a-ok
```console
$ terraform init
## 8< -- snip -- >8
$ terraform plan
data.aws_availability_zones.az_list: Refreshing state...

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

$ tflint --deep
Error: Failed to initialize a runner: main.tf:6,12-15: Variables not allowed; Variables may not be used here., and 1 other diagnostic(s)
```

Running `tflint --deep` with v0.9.2 does not report this error

**Comment by wata727** [comment]
Thanks for reporting this. This bug has been fixed in v0.10.1.
